receiver/carbonreceiver/transport/client/plaintext_client.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// This code was initially taken from
+// https://github.com/census-ecosystem/opencensus-go-exporter-graphite/tree/master/internal/client
+// and modified for the needs of testing the Carbon receiver package.
+
+// Graphite is a struct that defines the relevant properties of a graphite
+// connection
+type Graphite struct {
+	Host    string
+	Port    int
+	Timeout time.Duration
+	Conn    io.Writer
+}
+
+// defaultTimeout is the default number of seconds that we're willing to wait
+// before forcing the connection establishment to fail
+const defaultTimeout = 5
+
+// NewGraphite is a method that's used to create a new Graphite
+func NewGraphite(network, host string, port int) (*Graphite, error) {
+	graphite := &Graphite{Host: host, Port: port}
+	err := graphite.connect(network)
+	if err != nil {
+		return nil, err
+	}
+
+	return graphite, nil
+}
+
+// connect populates the Graphite.conn
+func (g *Graphite) connect(network string) error {
+	if cl, ok := g.Conn.(io.Closer); ok {
+		cl.Close()
+	}
+
+	address := fmt.Sprintf("%s:%d", g.Host, g.Port)
+	if g.Timeout == 0 {
+		g.Timeout = defaultTimeout * time.Second
+	}
+
+	var err error
+	var conn net.Conn
+
+	if network == "udp" {
```

Comment:
Please define network type constants in one central place.

Reply:
Done (creating enum to limit it to tcp and udp). The client didn't receive the same attention because it is intended to be used on tests only but it is public so different packages under carbonreceiver can use it. Likely I would do further changes later, eg: pass an endpoint instead of host+port etc.